src/checkers/inference/solver/backend/AbstractFormatTranslator.java

```diff
@@ -91,9 +94,10 @@ protected void postInit() {
         return null;
     }
 
-    /*Caution: if subclass overrides the serialize(Constraint) methods, nullness of the corresponding
-    * encoder field must be checked before dereferencing. Because subclass might not support some constraint
-    * encoding and the encoders of those unsupported constraints are by default null. */
+    /** Caution: if subclass overrides the serialize(XXXConstraint) methods, nullness of the corresponding encoder
+     * field must be checked before dereferencing unless you overrode corresponding createXXXConstraintEncoder to
+     * return a non-null value. Because subclass might not support some constraint encoding and the encoders of those
+     * unsupported constraints are by default null. By default, serialize(XXXConstraint) methods don't encode constraints*/
```

Comment:
This javadoc seems not specific to `serialize(SubtypeConstraint)`, maybe merge this one with the javadoc of this class?